### ISSUE TITLE
salt-bootstrap doesn't recognize nitrogen release on debian stretch

### ISSUE THREAD

**Issue by SebastianWendel** [saltstack/salt-bootstrap#1118]
### Description of Issue/Question
salt-bootstrap doesn't recognize nitrogen release on debian 9 alias stretch.  

It also does not use the release version from http://repo.saltstack.com/apt/debian/9/amd64/latest and instead installs the version from the system repo 2016.11. 

### Setup and Steps to Reproduce Issue
```
Vagrant.configure("2") do |config|
  config.vm.box = "debian/stretch64"
  config.vm.provider "virtualbox" do |vb|
  end
  config.vm.provision :salt do |salt|
    salt.install_master = true
  end
end
```

```
vagrant destroy -f && vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'debian/stretch64'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'debian/stretch64' is up to date...
==> default: Setting the name of the VM: salt-vagrant-stretch_default_1500976836919_80081
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: 
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default: 
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if it's present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
[default] No installation found.
...
...
...
...
==> default: Running provisioner: salt...
Checking if salt-minion is installed
salt-minion was not found.
Checking if salt-call is installed
salt-call was not found.
Checking if salt-master is installed
salt-master was not found.
Bootstrapping Salt... (this may take a while)
Salt successfully installed!
run_overstate set to false. Not running state.overstate.
run_highstate set to false. Not running state.highstate.
orchestrate is nil. Not running state.orchestrate.

==> default: Machine 'default' has a post `vagrant up` message. This is a message
==> default: from the creator of the Vagrantfile, and not from Vagrant itself:
==> default: 
==> default: Vanilla Debian box. See https://atlas.hashicorp.com/debian/ for help and bug reports
```

```
root@stretch:~# dpkg -l | grep salt
ii  salt-common                   2016.11.2+ds-1                 all          shared libraries that salt requires for all packages
ii  salt-master                   2016.11.2+ds-1                 all          remote manager to administer servers via salt
ii  salt-minion                   2016.11.2+ds-1                 all          client package for salt, the distributed remote execution system
```

```
root@stretch:~# /tmp/bootstrap_salt.sh stable 2017.7
+ [ -x /usr/bin/fetch ]
+ [ -x /usr/bin/curl ]
+ + sh -sx -- stable 2017.7
python -c import urllib; print urllib.urlopen("https://raw.githubusercontent.com/saltstack/salt-bootstrap/stable/bootstrap-salt.sh").read()
+ set -o nounset
+ __ScriptVersion=2017.05.24
+ __ScriptName=bootstrap-salt.sh
+ __ScriptFullName=sh
+ __ScriptArgs=stable 2017.7
+ BS_TRUE=1
+ BS_FALSE=0
+ __DEFAULT_SLEEP=3
+ tput colors
+ _COLORS=256
+ __detect_color_support
+ [ 0 -eq 0 ]
+ [ 256 -gt 2 ]
+ RC=\033[1;31m
+ GC=\033[1;32m
+ BC=\033[1;34m
+ YC=\033[1;33m
+ EC=\033[0m
+ _KEEP_TEMP_FILES=0
+ _TEMP_CONFIG_DIR=null
+ _SALTSTACK_REPO_URL=https://github.com/saltstack/salt.git
+ _SALT_REPO_URL=https://github.com/saltstack/salt.git
+ _DOWNSTREAM_PKG_REPO=0
+ _TEMP_KEYS_DIR=null
+ _SLEEP=3
+ _INSTALL_MASTER=0
+ _INSTALL_SYNDIC=0
+ _INSTALL_MINION=1
+ _INSTALL_CLOUD=0
+ _VIRTUALENV_DIR=null
+ _START_DAEMONS=1
+ _DISABLE_SALT_CHECKS=0
+ _ECHO_DEBUG=0
+ _CONFIG_ONLY=0
+ _PIP_ALLOWED=0
+ _PIP_ALL=0
+ _SALT_ETC_DIR=/etc/salt
+ _SALT_CACHE_DIR=/var/cache/salt
+ _PKI_DIR=/etc/salt/pki
+ _FORCE_OVERWRITE=0
+ _GENTOO_USE_BINHOST=0
+ _EPEL_REPO=epel
+ _EPEL_REPOS_INSTALLED=0
+ _UPGRADE_SYS=0
+ _INSECURE_DL=0
+ _CURL_ARGS=
+ _FETCH_ARGS=
+ _GPG_ARGS=
+ _WGET_ARGS=
+ _ENABLE_EXTERNAL_ZMQ_REPOS=0
+ _SALT_MASTER_ADDRESS=null
+ _SALT_MINION_ID=null
+ _SIMPLIFY_VERSION=1
+ _LIBCLOUD_MIN_VERSION=0.14.0
+ _EXTRA_PACKAGES=
+ _HTTP_PROXY=
+ _SALT_GIT_CHECKOUT_DIR=/tmp/git/salt
+ _NO_DEPS=0
+ _FORCE_SHALLOW_CLONE=0
+ _DISABLE_SSL=0
+ _DISABLE_REPOS=0
+ _CUSTOM_REPO_URL=null
+ _CUSTOM_MASTER_CONFIG=null
+ _CUSTOM_MINION_CONFIG=null
+ _QUIET_GIT_INSTALLATION=0
+ _REPO_URL=repo.saltstack.com
+ _PY_EXE=
+ _INSTALL_PY=0
+ getopts :hvnDc:g:Gyx:wk:s:MSNXCPFUKIA:i:Lp:dH:ZbflV:J:j:rR:aq opt
+ shift 0
+ [ 1 -eq 0 ]
+ [ 1 -eq 0 ]
+ [ 1 -eq 0 ]
+ [ null != null ]
+ [ null != null ]
+ [ 2 -eq 0 ]
+ __check_unparsed_options stable 2017.7
+ shellopts=stable 2017.7
+ [ -f /usr/xpg4/bin/grep ]
+ grep=grep
+ + grep -E (^|[[:space:]])[-]+[[:alnum:]]
echo stable 2017.7
+ unparsed_options=
+ [  !=  ]
+ ITYPE=stable
+ shift
+ echo stable+ egrep (stable|testing|daily|git)

+ [ stable =  ]
+ [ stable = git ]
+ [ stable = stable ]
+ [ 1 -eq 0 ]
+ __check_unparsed_options 2017.7
+ shellopts=2017.7
+ [ -f /usr/xpg4/bin/grep ]
+ grep=grep
+ + grep -E (^|[[:space:]])[-]+[[:alnum:]]
echo 2017.7
+ unparsed_options=
+ [  !=  ]
+ + egrep ^(latest|1\.6|1\.7|2014\.1|2014\.7|2015\.5|2015\.8|2016\.3|2016\.11)$
echo 2017.7
+ [  !=  ]
+ + egrep ^([0-9]*\.[0-9]*\.[0-9]*)$
echo 2017.7
+ [  !=  ]
+ echo Unknown stable version: 2017.7 (valid: 1.6, 1.7, 2014.1, 2014.7, 2015.5, 2015.8, 2016.3, 2016.11, latest, $MAJOR.$MINOR.$PATCH)
Unknown stable version: 2017.7 (valid: 1.6, 1.7, 2014.1, 2014.7, 2015.5, 2015.8, 2016.3, 2016.11, latest, $MAJOR.$MINOR.$PATCH)
+ exit 1
Traceback (most recent call last):
  File "<string>", line 1, in <module>
IOError: [Errno 32] Broken pipe
```

```
/tmp/bootstrap_salt.sh stable 2017.7
Unknown stable version: 2017.7 (valid: 1.6, 1.7, 2014.1, 2014.7, 2015.5, 2015.8, 2016.3, 2016.11, latest, $MAJOR.$MINOR.$PATCH)
Traceback (most recent call last):
  File "<string>", line 1, in <module>
IOError: [Errno 32] Broken pipe
```

### Versions and Systems
* Hostsystem Ubuntu 16.04
* Vagrant 1.9.7
* Hostdriver Virtualbox 5.1.22
* Guest Debian 9 Stretch

**Comment by rallytime** [comment]
Hi @sourceindex - this has been fixed on the `develop` branch in the [salt-bootstrap repo](https://github.com/saltstack/salt-bootstrap). That change will be available in the next release of the salt-bootstrap script or you can use the `develop` version of the script.

For your question about seeing the `Unknown stable version: 2017.7` - thank you for submitting the pull request https://github.com/saltstack/salt-bootstrap/pull/1120. I had that on my radar the other day, but I forgot to submit it. :)

For now, to get the `2017.7` packages, you can run the bootstrap script with just `stable` (no branch name after), or you don't need to put anything there at all and just run `sh bootstrap-salt.sh` as the latest stable packages are the default.

**Comment by SebastianWendel** [comment]
@rallytime thanks and no problem.

No argumens for `bootstrap-salt.sh` is my usual case, `2017.7` was just for testing when i experienced the issue.

Do you have any timeframe for the stable release?

**Comment by rallytime** [comment]
@sourceindex I was just talking to @Ch3LL about that. We want to get a new stable cut for salt-bootstrap in time for it to be included in the 2017.7.2 release of Salt.
